### PR TITLE
Correctly document TimeSeries peek() args/kwargs

### PR DESCRIPTION
**Pull request by dstansby**
Sort of xref #2420. This

- Tidies and clarifies docstrings for all the `peek()` methods in timeseries sources.
- For `noaa` does a check for plot type, and errors if it's not a recognised type
- A couple of small PEP8 fixes

At some point we should standardise these (ie. always have `**plot_kwargs), but this just makes the docs align with the code.

### CODE REVIEW

**Comment by wafels** [comment]
I posted a new issue #3426.  Since a number of the files you are changing already are affected by #3426, it might be worth fixing that too in this PR (it's a super-easy fix).

File: sunpy/timeseries/sources/noaa.py [hunk 1]
```diff
@@ -205,7 +207,7 @@ def peek(self, **plot_args):
         Parameters
         ----------
         **plot_args : `dict`
-            Any additional plot arguments that should be used when plotting.
+            Unused.
```
[the remark this answers is not in the document]
Comment (dstansby) [reply sunpy/timeseries/sources/noaa.py]: Yes, but I've limited this PR is to document what exists, not change any API at all.

File: changelog/3378.breaking.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+:meth:`NOAAIndicesTimeSeries.peek()` now checks that the `type` argument is a
```
Comment (Cadair) [inline changelog/3378.breaking.rst]:
```suggestion
`NOAAIndicesTimeSeries.peek` now checks that the `type` argument is a
```
?

File: sunpy/timeseries/sources/noaa.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hayesla) [reply sunpy/timeseries/sources/noaa.py]: Hmm yes its weird that some peek() take kwargs and some don't - I'm of the opinion that they probably shouldn't - I'll open up an issue regarding this

File: sunpy/timeseries/sources/eve.py [hunk 3]
```diff
@@ -61,6 +61,15 @@ class ESPTimeSeries(GenericTimeSeries):
 
     @peek_show
     def peek(self, title='EVE/ESP Level1', **kwargs):
+        """
+        Parameters
+        ----------
+        title : str, optional
```
Comment (nabobalis) [inline sunpy/timeseries/sources/eve.py]:
```suggestion
        title : `str`, optional
```

File: sunpy/timeseries/sources/eve.py [hunk 4]
```diff
@@ -61,6 +61,15 @@ class ESPTimeSeries(GenericTimeSeries):
...
+            Plot title.
+        **kwargs : dict
```
Comment (nabobalis) [inline sunpy/timeseries/sources/eve.py]:
```suggestion
        **kwargs : `dict`
```

File: sunpy/timeseries/sources/eve.py [hunk 5]
```diff
@@ -61,6 +61,15 @@ class ESPTimeSeries(GenericTimeSeries):
...
+            Additional plot keyword arguments that are handed to
+            :meth:`pandas.DataFrame.plot`.
```
Comment (nabobalis) [inline sunpy/timeseries/sources/eve.py]: Do we need to add `:meth:`?

**Pull request merged**